alicloud/data_source_alicloud_disks.go

```diff
@@ -0,0 +1,259 @@
+package alicloud
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"regexp"
+	"fmt"
+	"log"
+)
+
+func dataSourceAlicloudDisks() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudDisksRead,
+
+		Schema: map[string]*schema.Schema{
+			"ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+				MinItems: 1,
+			},
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+				ForceNew:     true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"category": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"encrypted": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
```

Comment:
I think set `encrypted` to `TypeBool` will be better.

Reply:
I did that at the beginning, but I need 3 values: not-set (results are not filtered), true, false.
This is because of the following code:
```
v, ok := d.GetOk("encrypted")
```
If `encrypted` is TypeBool and set to false, then `ok` contains false, which is wrong (`ok` should contain false only if I didn't set `encrypted`).

Only using `on` and `off` allowed me to have 3 possible values.

Reply:
If `encrypted` allow "on" or "off", it should have a ValidateFunc to limit it: `validateAllowedStringValue`.